### PR TITLE
some test coverage for Value::iterator

### PR DESCRIPTION
**Pull request by BillyDonahue**

### CODE REVIEW

**Comment by BillyDonahue** [comment]
Brings json_valueiterator.inl to 100% coverage.

File: src/test_lib_json/main.cpp [hunk 1]
```diff
@@ -3394,18 +3395,85 @@ JSONTEST_FIXTURE_LOCAL(BuilderTest, settings) {
...
+    auto it = json.begin();
+    for (;; ++it, ++i) {
+      auto dist = it - json.begin();
+      JSONTEST_ASSERT_EQUAL(i, dist);
+      if (it == json.end())
+        break;
+    }
+  }
+  {
+    Json::Value j; // empty
```
Comment (dota17) [inline src/test_lib_json/main.cpp]: Can we directly use `Json::Value empty;`?

File: src/test_lib_json/main.cpp [hunk 2]
```diff
@@ -3394,18 +3395,85 @@ JSONTEST_FIXTURE_LOCAL(BuilderTest, settings) {
...
+}
+
+JSONTEST_FIXTURE_LOCAL(IteratorTest, nullValues) {
+  {
+    Json::Value json;
+    auto end = json.end();
+    auto endCopy = end;
+    JSONTEST_ASSERT(endCopy == end);
+    endCopy = end;
+    JSONTEST_ASSERT(endCopy == end);
```
Comment (dota17) [inline src/test_lib_json/main.cpp]: I don't understand why repeatly assert this.

Comment (BillyDonahue) [reply src/test_lib_json/main.cpp]: I'm testing how the copy assignment operator= functions with `null` iterators. So the first assert is checking that copy construction yields equal iterators. The second is checking that copy assignment yields equal iterators as well.

File: src/test_lib_json/main.cpp (hunk 1 above)
Comment (BillyDonahue) [reply src/test_lib_json/main.cpp]: Done.

**Pull request merged**